src-electron/db/query-util.js

```diff
@@ -83,5 +83,72 @@ function sqlQueryForDataTypeByNameAndClusterId(
   return clusterId ? queryWithClusterId : queryWithoutClusterId
 }
 
+/**
+ * Formulate a sqlite query string for a data type from the given cluster name and package IDs.
+ * @param {*} typeDiscriminator
+ * @param {*} name data type name
+ * @param {*} clusterName
+ * @param {*} packageIds
+ * @param {*} options
+ * @returns SQLite query string
+ */
+function sqlQueryForDataTypeByNameAndClusterName(
+  typeDiscriminator,
+  name,
+  clusterName,
+  packageIds,
+  options = {}
+) {
+  //let clusterNameWithNoSpaces = clusterName ? clusterName.replace(/\s+/g, '') : clusterName;
+  let typeTableName = typeDiscriminator.toUpperCase()
+  let numberExtensionString =
+    typeDiscriminator == 'number' ? 'NUMBER.IS_SIGNED, ' : ''
+  let checkLowerCaseString =
+    typeDiscriminator != 'number' && typeDiscriminator != 'struct'
+      ? `OR DATA_TYPE.NAME = '${name}'`
+      : ''
+  let structExtensionString =
+    typeDiscriminator == 'struct'
+      ? 'STRUCT.IS_FABRIC_SCOPED, DATA_TYPE.DISCRIMINATOR_REF, '
+      : ''
+  let selectQueryString = `
+  SELECT
+    ${typeTableName}.${typeTableName}_ID,
+    ${structExtensionString}
+    DATA_TYPE.NAME AS NAME,
+    ${numberExtensionString}
+    ${typeTableName}.SIZE AS SIZE,
+    CLUSTER.NAME AS CLUSTER_NAME,
+    REPLACE(CLUSTER.NAME, ' ', '') AS CLUSTER_NAME_WITHOUT_SPACES
```

Comment:
I see why you ended up here, but sadly it's not just a "without spaces", though maybe in practice it is: what comes into the Darwin code there is `(asUpperCamelCase clusterName preserveAcronymes=true)` or so.

I don't have any better ideas than this, though this seems extremely unexpected as a caller.  :(  Maybe it really would be better to change all consumers of asObjectiveCType/Class to pass in a cluster id instead...  That's 60-ish callsites, but maybe we can manage that....  

Or of course we could leave this bit, which is obviously less work for me, but might come back to bite us sometime.

Reply:
We should still keep this just to satisfy an old chip release and the latest zap but eventually changing to cluster IDs is safer.